### PR TITLE
Fast follow dataset cleanups

### PR DESCRIPTION
**Pull request by prithvikannan**
## Related Issues/PRs

https://github.com/mlflow/mlflow/pull/8186

## What changes are proposed in this pull request?

Some small code cleanups.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/data/delta_dataset_source.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def _resolve(cls, raw_source: str) -> DeltaDatasetSourceType:
     def _is_databricks_uc_table(self):
         if is_in_databricks_runtime() and self._delta_table_name is not None:
             try:
-                catalog_name, _, _ = self._delta_table_name.split(".")
+                catalog_name = self._delta_table_name.split(".", 1)[0]
```
Comment (harupy) [inline mlflow/data/delta_dataset_source.py:87]: Do we still need try-catch?

File: mlflow/data/filesystem_dataset_source.py [hunk 2]
```diff
@@ -1,15 +1,10 @@
...
-from typing import TypeVar, Any, Dict
+from typing import Any, Dict
 
 from mlflow.data.dataset_source import DatasetSource
 from mlflow.utils.annotations import experimental
 
 
-FileSystemDatasetSourceType = TypeVar(
-    "FileSystemDatasetSourceType", bound="FileSystemDatasetSource"
-)
```
Comment (harupy) [inline mlflow/data/filesystem_dataset_source.py:10]: Other datasets also contain this. Can we update them as well?

File: mlflow/data/http_dataset_source.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def load(self, dst_path=None) -> str:
         content_disposition = resp.headers.get("Content-Disposition")
         if (
             content_disposition is not None
-            and len(file_names := re.findall("filename=(.+)", content_disposition)) > 0
+            and len(file_names := re.findall(r"filename=(.+)", content_disposition)) > 0
         ):
             # NB: If the filename is quoted, unquote it
             basename = file_names[0].lstrip('"').rstrip('"').lstrip("'").rstrip("'")
```
Comment (harupy) [inline mlflow/data/http_dataset_source.py:63]:
```
basename = file_names[0].strip('\'"')
```

Can this line be simplified?

File: mlflow/data/http_dataset_source.py [hunk 4]
```diff
@@ -57,7 +57,7 @@ def load(self, dst_path=None) -> str:
         content_disposition = resp.headers.get("Content-Disposition")
         if (
             content_disposition is not None
-            and len(file_names := re.findall("filename=(.+)", content_disposition)) > 0
+            and len(file_names := re.findall(r"filename=(.+)", content_disposition)) > 0
```
Comment (harupy) [inline mlflow/data/http_dataset_source.py:60]:
```suggestion
            and filename := next(re.finditer(r"filename=(.+)", content_disposition), None))
```

nit

**Review by harupy — APPROVED** [review]
Left some comments, LGTM once they are addressed!

File: mlflow/data/delta_dataset_source.py (hunk 1 above)
Comment (prithvikannan) [reply mlflow/data/delta_dataset_source.py:87]: no :)

File: mlflow/data/http_dataset_source.py [hunk 5]
```diff
@@ -55,12 +52,11 @@ def load(self, dst_path=None) -> str:
...
         content_disposition = resp.headers.get("Content-Disposition")
-        if (
-            content_disposition is not None
-            and len(file_names := re.findall("filename=(.+)", content_disposition)) > 0
+        if content_disposition is not None and (
+            file_name := next(re.finditer(r"filename=(.+)", content_disposition), None)
         ):
             # NB: If the filename is quoted, unquote it
-            basename = file_names[0].lstrip('"').rstrip('"').lstrip("'").rstrip("'")
+            basename = file_name[1].strip("'\"")
```
Comment (prithvikannan) [inline mlflow/data/http_dataset_source.py:59]: need to use 1 here bc `file_name` is a <re.Match object>

**Pull request merged**